homeassistant/components/purpleair/config_flow.py

```diff
@@ -0,0 +1,220 @@
+"""Config flow for PurpleAir integration."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any
+
+from aiopurpleair import API
+from aiopurpleair.errors import InvalidApiKeyError, PurpleAirError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_validation as cv
+from homeassistant.helpers.selector import (
+    SelectOptionDict,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_SENSOR_INDICES, DOMAIN, LOGGER
+
+CONF_DISTANCE = "distance"
+CONF_NEARBY_SENSOR_OPTIONS = "nearby_sensor_options"
+CONF_SENSOR_INDEX = "sensor_index"
+
+DEFAULT_DISTANCE = 5
+
+API_KEY_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_API_KEY): cv.string,
+    }
+)
+
+
+@callback
+def async_get_api(hass: HomeAssistant, api_key: str) -> API:
+    """Get an aiopurpleair API object."""
+    session = aiohttp_client.async_get_clientsession(hass)
+    return API(api_key, session=session)
+
+
+@callback
+def async_get_coordinates_schema(hass: HomeAssistant) -> vol.Schema:
+    """Define a schema for searching for sensors near a coordinate pair."""
+    return vol.Schema(
+        {
+            vol.Inclusive(
+                CONF_LATITUDE, "coords", default=hass.config.latitude
+            ): cv.latitude,
+            vol.Inclusive(
+                CONF_LONGITUDE, "coords", default=hass.config.longitude
+            ): cv.longitude,
+            vol.Optional(CONF_DISTANCE, default=DEFAULT_DISTANCE): cv.positive_int,
+        }
+    )
+
+
+@callback
+def async_get_nearby_sensors_schema(options: list[SelectOptionDict]) -> vol.Schema:
```

Comment:
Is there a reason we don't inline the data schema in the form call? It's only used once, right?

Reply:
Now that the options flow has been merged, it's used in two places.